src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -634,34 +625,56 @@ private static BlittableJsonReaderObject RevertSnapshotFlags(DocumentsOperationC
+            RevisionsCollectionConfiguration configuration, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks,
+            bool fromDelete = false)
         {
             using (GetKeyPrefix(context, lowerId, out Slice prefixSlice))
             {
                 // We delete the old revisions after we put the current one,
                 // because in case that MinimumRevisionsToKeep is 3 or lower we may get a revision document from replication
                 // which is old. But because we put it first, we make sure to clean this document, because of the order to the revisions.
                 var revisionsCount = IncrementCountOfRevisions(context, prefixSlice, 1);
-                DeleteOldRevisions(context, table, prefixSlice, collectionName, configuration, revisionsCount, nonPersistentFlags, changeVector, lastModifiedTicks);
+                DeleteOldRevisions(context, table, prefixSlice, collectionName, configuration, revisionsCount, nonPersistentFlags, changeVector, lastModifiedTicks, out _, deletedDoc: fromDelete);
             }
         }
 
         private bool DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, CollectionName collectionName,
-            RevisionsCollectionConfiguration configuration, long revisionsCount, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks)
+            RevisionsCollectionConfiguration configuration, long revisionsCount, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks, out long currentRevisionsCount, bool deletedDoc = false, bool skipForceCreated = false)
         {
+            currentRevisionsCount = revisionsCount;
+
             var moreRevisionToDelete = false;
             if (nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromSmuggler))
                 return false;
 
             if (nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromReplication))
                 return false;
 
-            if (configuration.MinimumRevisionsToKeep.HasValue == false &&
-                configuration.MinimumRevisionAgeToKeep.HasValue == false)
-                return false;
-
             long numberOfRevisionsToDelete;
-            if (configuration.MinimumRevisionsToKeep.HasValue)
+            if (deletedDoc && configuration.PurgeOnDelete)
+            {
+                numberOfRevisionsToDelete = long.MaxValue;
+            }
+            else if (configuration == ConflictConfiguration.Default || 
+                     configuration == ZeroConfiguration)
+            {
+                if(deletedDoc && ConflictConfiguration.Default.PurgeOnDelete)
+                    numberOfRevisionsToDelete = long.MaxValue;
+                else
+                {
+                    var handleNotConflicted = nonPersistentFlags.Contain(NonPersistentDocumentFlags.ByEnforceRevisionConfiguration);
```

Comment:
When you have only the 'Conflict Revisions Configuration' and you perform `EnforceConfiguration`,
you delete the conflicted/resolved by their 'Conflict Revisions Configuration' and **all** the rest (non-conflicted) revisions will be deleted because they don't have any config (except the Force-Created if you don't want to delete them).

When you have only the 'Conflict Revisions Configuration' and you have some non-conflicted revisions, and you change the document - you delete the conflicted/resolved by their 'Conflict Revisions Configuration' and **all** the rest (non-conflicted) revisions not touched.

Reply:
so we have 3 types of revisions (deleted or not):
1. regular
2. conflicted
3. force created

Let be more explicit on what we plan to handle.
and create some sort of an `enum` with flags of what we want to handle and replace all of those `skipForceCreated` / `handleNotConflicted`, etc...

Reply:
Done